### PR TITLE
Feat create home

### PR DESCRIPTION
**Pull request by lMahesvara**
Changes made: 

- Sidebar with full course list and navigation.
- Home navbar with user dropdown.
- Mobile menu.
- Added skeletons for components with async data loading.

### CODE REVIEW

File: src/components/ListOfCoursesCS.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { getCourses } from '@/services/api'
```
Comment (fcampoa) [inline src/components/ListOfCoursesCS.tsx:1]: Que significan las CS al final del nombre?

File: src/components/ListOfCoursesSS.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { getCourses } from '@/services/api'
```
Comment (fcampoa) [inline src/components/ListOfCoursesSS.tsx:1]: Que significa SS al final del nombre?

**Review by fcampoa — CHANGES_REQUESTED** [review]

File: src/components/ListOfCoursesCS.tsx (hunk 1 above)
Comment (lMahesvara) [reply src/components/ListOfCoursesCS.tsx:1]: Client side

File: src/components/ListOfCoursesSS.tsx (hunk 2 above)
Comment (lMahesvara) [reply src/components/ListOfCoursesSS.tsx:1]: Server side

**Pull request merged**